### PR TITLE
[release-v1.8] Apply patches to make OCP 4.8 -> 4.11 work

### PR DESCRIPTION
**Pull request by creydr**
Currently  `48-test-e2e-aws-ocp-48` test is failing, because the eventing-operator does not come up cause of `failed to apply non rbac manifest: no matches for kind "HorizontalPodAutoscaler" in version "autoscaling/v2"`

https://github.com/openshift/knative-eventing/pull/1928 addressed it and set the apiVersion for HPA to autoscaling/v2beta2. This PR cherry-picks these changes into the 1.8 branch.

### CODE REVIEW

**Comment by matzew** [comment]
Please run `make RELEASE=v1.8 generate-release ` on your PR, @creydr

**Comment by creydr** [comment]
> Please run `make RELEASE=v1.8 generate-release ` on your PR, @creydr

Done

**Comment by creydr** [comment]
waiting for https://github.com/openshift-knative/eventing/pull/61

**Comment by aliok** [comment]
/hold

Waiting until https://github.com/openshift-knative/eventing/pull/61 is merged

**Comment by matzew** [comment]
rebased and unholding, after 61 got merged (manually)

/unhold

**Comment by creydr** [comment]
@matzew @aliok could you PTAL?
Thanks in advance

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
